pysmt/solvers/yices.py

```diff
@@ -306,97 +322,99 @@ def convert(self, formula):
 
     def _check_term_result(self, res):
         if res == -1:
-            err = yicespy.yices_error_string()
+            err = yices_api.yices_error_string()
             raise InternalSolverError("Yices returned an error: " + err)
 
     def walk_and(self, formula, args, **kwargs):
-        res = yicespy.yices_and(len(args), args)
+        values = (yices_api.term_t * len(args))(*args)
+        res = yices_api.yices_and(len(args), values)
         self._check_term_result(res)
         return res
 
     def walk_or(self, formula, args, **kwargs):
-        res = yicespy.yices_or(len(args), args)
+        values = (yices_api.term_t * len(args))(*args)
```

Comment:
I am not sure what is going on here, can you help me understand? Is this allocating a list with the values of `args`?

Reply:
Exactly, they use ctypes and in this way we create a C-array containing all the args. Maybe we can wrap this in a utility function?

Reply:
Thanks. I don't think we need to do anything more here, it was just a bit surprising as the syntax is odd.